src/main/java/net/minecraftforge/common/ForgeConfigSpec.java

```diff
@@ -635,15 +635,27 @@ public void addComment(String value)
             // Don't use `validate` because it throws IllegalStateException, not NullPointerException
             Preconditions.checkNotNull(value, "Passed in null value for comment");
 
-            if (value.isBlank() && comment.size() == 0)
-                throw new IllegalArgumentException("Can not add zero-length comment as top-most comment");
-
             comment.add(value);
         }
 
         public void clearComment() { comment.clear(); }
         public boolean hasComment() { return this.comment.size() > 0; }
-        public String buildComment() { return LINE_JOINER.join(comment); }
+        public String buildComment() { return buildComment(List.of("unknown", "unknown")); }
+        public String buildComment(final List<String> path)
+        {
+            if (comment.stream().allMatch(String::isBlank))
+            {
+                if (FMLEnvironment.production)
+                    LogManager.getLogger().warn(CORE, "Detected a comment that is all whitespace for config option {}, which causes obscure bugs in Forge's config system and will cause a crash in the future. Please report this to the mod author.",
+                            DOT_JOINER.join(path));
```

Comment:
Not only does this irk me, but also that this is in line with how other parts of the class get the logger instance.

For the sake of better code, please refactor all uses of `LogManager.getLogger()` within this class to refer to a constant which is initialized with `LogManager.getLogger()` instead.

Also, noting for the future: we need to look at the present logging markers, specifically and especially `CORE`, and write down what each marker refers to.